cmake/build_config.h.in

```diff
@@ -0,0 +1,28 @@
+/*!
+ * Copyright 2019 by Contributors
+ * \file build_config.h
+ *
+ * Generated from `cmake/build_config.h.in` by cmake.
+ */
+#ifndef XGBOOST_BUILD_CONFIG_H_
+#define XGBOOST_BUILD_CONFIG_H_
+
+// These check are for Makefile.
+#if !defined(XGBOOST_MM_PREFETCH_PRESENT) && !defined(XGBOOST_BUILTIN_PREFETCH_PRESENT)
```

Comment:
I think I am missing something but how is it part of global version function?

Reply:
@trams No.  It's the `build_config.h` in include directory.  Since version info is also defined here so I copied it as cmake template.